src/feature_annotation/highly_variable_features_scanpy/script.py

```diff
@@ -130,11 +140,19 @@
 # call function
 try:
     out = sc.pp.highly_variable_genes(**hvg_args)
-    if par["obs_batch_key"] is not None:
+    if par["var_input"] is not None:
+        out.index = data[:, data.var["common_vars"]].var.index
```

Comment:
should this `data.var["common_vars"]` be `data.var[par["var_input"]]` (or `data.var[par["input_var_subset"]]` if the other suggestion is accepted) ?